packages/ember-application/lib/system/application-instance.js

```diff
@@ -212,6 +298,255 @@ let ApplicationInstance = EmberObject.extend(RegistryProxy, ContainerProxy, {
+    }
+
+    /**
+      Disable rendering completely.
+
+      When this flag is set to `true`, it will disable the entire rendering
+      pipeline. Essentially, this puts the app into "routing-only" mode. No
+      templates will be rendered, and no Components will be created.
+
+      @property render
+      @type boolean
+      @default true
+      @public
+    */
+    if (options.render !== undefined) {
+      this.render = !!options.render;
+    } else {
+      this.render = true;
+    }
+
+    if (!this.render) {
+      this.jQuery = null;
+      this.interactive = false;
+    }
+
+    /**
+      If present, render into the given `Document` object instead of the
+      global `window.document` object.
+
+      In practice, this is only useful in non-browser environment or in
+      non-interactive mode, because Ember's `jQuery` dependency is
+      implicitly bound to the current document, causing event delegation
+      to not work properly when the app is rendered into a foreign
+      document object (such as an iframe's `contentDocument`).
+
+      In non-browser mode, this could be a "`Document`-like" object as
+      Ember only interact with a small subset of the DOM API in non-
+      interactive mode. While the exact requirements have not yet been
+      formalized, the `SimpleDOM` library's implementation is known to
+      work.
+
+      @property document
+      @type Document
+      @default the global `document` object
+      @public
+    */
+    if (options.document) {
+      this.document = options.document;
+    } else {
+      this.document = document;
+    }
+
+    /**
+      If present, overrides the application's `rootElement` property on
+      the instance. This is useful for testing environment, where you
+      might want to append the root view to a fixture area.
+
+      In non-browser mode, because Ember does not have access to jQuery,
+      this options must be specified as a DOM `Element` object instead of
+      a selector string.
+
+      See the documentation on `Ember.Applications`'s `rootElement` for
+      details.
+
+      @property rootElement
+      @type String|Element
+      @default null
+      @public
+     */
+    if (options.rootElement) {
+      this.rootElement = options.rootElement;
+    }
+
+    /**
+      If present, overrides the router's `location` property with this
+      value. This is useful for environments where trying to modify the
+      URL would be inappropriate.
+
+      @property location
+      @type string
+      @default null
+      @public
+    */
+    if (options.location !== undefined) {
+      this.location = options.location;
+    }
+
+    if (options.jQuery !== undefined) {
+      this.jQuery = options.jQuery;
+    }
+
+    if (options.interactive !== undefined) {
+      this.interactive = !!options.interactive;
+    }
+  };
+
+  BootOptions.prototype.toEnvironment = function() {
+    let env = assign({}, environment);
+    env.hasDOM = this.browser;
```

Comment:
Like above, using `browser` (a noun) to mean a Boolean is very confusing to a reader unfamiliar with the code. I think `isBrowser` is a better choice.